autograd/tensor.py

```diff
@@ -210,6 +234,105 @@ def __neg__(self):
+        """
+        Compute the sum of tensor elements
+
+        params:
+            axis (int or tuple of ints, optional): Axis or axes along which a sum is performed. The default, axis=None, sums all of the elements of the input tensor.
+            keepdims (bool, optional): If this is set to True, the axes which are reduced are left in the result as dimensions with size one. With this option, the result will broadcast correctly against the input tensor.
+        For example:
+            - original tensor shape(3,4,5), axis(1,2), keepdims(True) -> result shape(3,1,1)
+            - original tensor shape(3,4,5), axis(1,2), keepdims(False) -> result shape(3,)
+            - original tensor shape(3,4,5), axis(None), keepdims(True) -> result shape(1,)
+            - original tensor shape(3,4,5), axis(None), keepdims(False) -> result shape()
+        """
+        # Handle scalar case
+        if not hasattr(self.data, "ndim") or self.data.ndim == 0:
+            return Tensor(
+                data=self.data, prev=(self,), requires_grad=self.requires_grad
+            )
+
+        # Normalize axis
+        axis = (axis,) if isinstance(axis, int) else axis
+
+        # Compute sum
+        result = Tensor(
+            data=np.sum(self.data, axis=axis, keepdims=keepdims),
+            prev=(self,),
+            requires_grad=self.requires_grad,
+        )
+
+        result._backward = lambda: self._reduce_ops_backward(result, axis, keepdims)
+        return result
+
+    def mean(self, axis=None, keepdims=False):
+        """
+        Compute the mean of tensor elements
+
+        params:
+            axis (int or tuple of ints, optional): Axis or axes along which a mean is performed. The default, axis=None, averages all of the elements of the input tensor.
+            keepdims (bool, optional): If this is set to True, the axes which are reduced are left in the result as dimensions with size one. With this option, the result will broadcast correctly against the input tensor.
+        For example:
+            - original tensor shape(3,4,5), axis(1,2), keepdims(True) -> result shape(3,1,1)
+            - original tensor shape(3,4,5), axis(1,2), keepdims(False) -> result shape(3,)
+            - original tensor shape(3,4,5), axis(None), keepdims(True) -> result shape(1,)
+            - original tensor shape(3,4,5), axis(None), keepdims(False) -> result shape()
+        """
+        # Normalize axis to a tuple
+        axis = (axis,) if isinstance(axis, int) else axis
+
+        # Create result tensor
+        result = Tensor(
+            data=np.mean(self.data, axis=axis, keepdims=keepdims),
+            prev=(self,),
+            requires_grad=self.requires_grad,
+        )
+
+        result._backward = lambda: self._reduce_ops_backward(
+            output=result, axis=axis, keepdims=keepdims
+        )
+        return result
+
+    def _reduce_ops_backward(self, output, axis=None, keepdims=False):
```

Comment:
Creating a general backward function for reduce ops, since they typically share the same backward behavior with the element count used to scale the gradient value, and special handling for `axis` and `keepdims` cases.